### PR TITLE
PP-682 Loading worldpay bin ranges on application startup

### PR DESCRIPTION
**Pull request by imranbohoran**
## WHAT
_A brief description of the pull request:_
**First up apologies for a big change set on this PR**.
* This PR consists of the main components for storing card data and retrieving them. 
* We have experimented with a Trie based data structure for storing data and calculated that we'd have over a million keys that needs to be artificially generated and inserted based on the card ranges we have from worldpay and discover.
* A benchmark has also been introduced to run against a test data set (but could also run against the real data set). As per the benchmark, running against the worldpay dataset, search time is around 75ms. 
* A dropwizard Managed object is used to load the data when the application startsup.

## HOW 
_Steps to test or reproduce:_
- Apart from reviewing the code, please run the benchmarks as well. You can run benchmark by simply running the main of the `InfinispanCardInformationStoreBenchmark`

## WHO
_Who should review this pull request:_

- [x] Developers
- [ ] WebOps

### CODE REVIEW

File: src/main/java/uk/gov/pay/card/db/loader/WorldpayBinRangeLoader.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    public WorldpayBinRangeLoader(String fileWithBINRanges) {
+        this.fileWithBINRanges = fileWithBINRanges;
+    }
+
+    @Override
+    public void loadDataTo(CardInformationStore store) throws DataLoaderException {
+        logger.info("Setting up card information store");
+
+        BufferedReader bufferedReader = getFileReader(fileWithBINRanges);
```
Comment (rauligs) [inline src/main/java/uk/gov/pay/card/db/loader/WorldpayBinRangeLoader.java]: minor: Use try with resources

**Comment by rauligs** [comment]
Looks good to me 👍  Only one comment (minor), happy to merge

File: src/main/java/uk/gov/pay/card/db/loader/WorldpayBinRangeLoader.java (hunk 1 above)
Comment (imranbohoran) [inline src/main/java/uk/gov/pay/card/db/loader/WorldpayBinRangeLoader.java]: Done

**Comment by imranbohoran** [comment]
rebuild

**Comment by imranbohoran** [comment]
rebuild

**Comment by imranbohoran** [comment]
rebuild

**Pull request merged**